### PR TITLE
upgrade(fleet): Plug rudimentary RC client for policies

### PR DESCRIPTION
**Pull request by BaptisteFoy**
### What does this PR do?
Plug rudimentary RC client for policies while we wait for the CDN

### CODE REVIEW

File: pkg/fleet/internal/cdn/cdn.go [hunk 1]
```diff
@@ -36,38 +56,125 @@ func New(env *env.Env) *CDN {
...
 	}
-	serializedConfig, err := config.Marshal()
+	service.Start()
+	defer func() { _ = service.Stop() }()
+	// Force a cache bypass
+	cfgs, err := service.ClientGetConfigs(ctx, &pbgo.ClientGetConfigsRequest{
+		Client: &pbgo.Client{
+			Id:            uuid.New().String(),
+			Products:      []string{"AGENT_CONFIG"},
+			IsUpdater:     true,
```
Comment (timothyalexandersoftware) [inline pkg/fleet/internal/cdn/cdn.go:108]: I never really noticed this field before, but it looks like you're doing updater client tracking separately. Do you need/expect client tracking for the CDN clients like the updater when it fetches the catalog/policies? Or do you only need client tracking from fetching non-CDN agent-based configs?

File: pkg/fleet/internal/cdn/cdn.go [hunk 2]
```diff
@@ -36,38 +56,125 @@ func New(env *env.Env) *CDN {
...
+		return nil, err
 	}
-	serializedConfig, err := config.Marshal()
+	service.Start()
+	defer func() { _ = service.Stop() }()
+	// Force a cache bypass
+	cfgs, err := service.ClientGetConfigs(ctx, &pbgo.ClientGetConfigsRequest{
+		Client: &pbgo.Client{
+			Id:            uuid.New().String(),
+			Products:      []string{"AGENT_CONFIG"},
```
Comment (timothyalexandersoftware) [inline pkg/fleet/internal/cdn/cdn.go:107]: is `AGENT_CONFIG` going to be synced to the cdn?

File: pkg/fleet/internal/cdn/config.go [hunk 3]
```diff
@@ -16,33 +19,85 @@ const (
...
-type config map[interface{}]interface{}
-
 // layer is a config layer that can be merged with other layers into a config.
 type layer struct {
-	ID      string
-	Content map[interface{}]interface{}
+	ID                  string                 `json:"name"`
+	AgentConfig         map[string]interface{} `json:"config"`
+	SecurityAgentConfig map[string]interface{} `json:"security_agent"`
+	SystemProbeConfig   map[string]interface{} `json:"system_probe"`
```
Comment (timothyalexandersoftware) [inline pkg/fleet/internal/cdn/config.go:27]: Just an FYI that I believe this is being updated soon to also support ClusterAgent configs. Not sure if the updater will be responsible for installing/configuring the cluster agent or not https://github.com/DataDog/datadog-operator/pull/1340/commits/51d2cf97370b4a5278df24135b119e4816e96415#diff-063b72ecc49939a1029c3d3820e85bf906aae54a38203c49563a0699fc0cce69R65

File: pkg/fleet/internal/cdn/cdn.go [hunk 4]
```diff
@@ -36,38 +56,125 @@ func New(env *env.Env) *CDN {
...
 	}
-	config, err := newConfig(baseLayer, overrideLayer)
+	service, err := remoteconfig.NewService(
+		config,
+		"Datadog Installer",
+		fmt.Sprintf("https://config.%s", c.env.Site),
+		hostname,
+		getHostTags(ctx, config),
+		&rctelemetryreporterimpl.DdRcTelemetryReporter{}, // No telemetry for this client
+		"9.9.9",
```
Comment (coignetp) [inline pkg/fleet/internal/cdn/cdn.go:95]: **🥜 nitpick:** ‏Let's change it now, or we'll forget to update it

**Review by moth-a — APPROVED** [review]
Looks fine to me as a temporary workaround, but I left a few questions for myself to better understand the updater as an RC client

File: pkg/fleet/internal/cdn/cdn.go [hunk 5]
```diff
@@ -36,38 +56,125 @@ func New(env *env.Env) *CDN {
...
+		options...,
+	)
 	if err != nil {
-		return Config{}, err
+		return nil, err
 	}
-	serializedConfig, err := config.Marshal()
+	service.Start()
+	defer func() { _ = service.Stop() }()
```
[the remark this answers is not in the document]
Comment (BaptisteFoy) [reply pkg/fleet/internal/cdn/cdn.go:102]: Yes, this is running before the agent install

File: pkg/fleet/internal/cdn/config.go [hunk 6]
```diff
@@ -16,33 +19,85 @@ const (
...
+
+		if l.SystemProbeConfig != nil {
+			systemProbeAgentConfig, err := merge(mergedLayer.SystemProbeConfig, l.SystemProbeConfig)
+			if err != nil {
+				return nil, err
+			}
+			mergedLayer.SystemProbeConfig = systemProbeAgentConfig.(map[string]interface{})
+		}
+	}
+	mergedLayer.AgentConfig[layerKeys] = layerIDs
```
Comment (coignetp) [inline pkg/fleet/internal/cdn/config.go:66]: **❓ question:** ‏Is this to report the applied layers? Let's add a description around it

File: pkg/fleet/internal/cdn/config.go (hunk 3 above)
Comment (BaptisteFoy) [reply pkg/fleet/internal/cdn/config.go:27]: Thanks for the FYI. That won't be necessary with the current state of things: everything k8s should be managed either by the operator or a new component. 

This installer is host-only

File: pkg/fleet/internal/cdn/cdn.go (hunk 2 above)
Comment (BaptisteFoy) [reply pkg/fleet/internal/cdn/cdn.go:107]: Yes, as the Fleet Policies config content will be on `AGENT_CONFIG` we need them at startup

File: pkg/fleet/internal/cdn/cdn.go (hunk 1 above)
Comment (BaptisteFoy) [reply pkg/fleet/internal/cdn/cdn.go:108]: It's not a defined need right now. We'll have tracking needs to send an AGENT_TASK "trigger policy experiment" config once we detect config drift (between AGENT_CONFIG received & what's actually applied), but that will likely go through the regular client for AGENT_TASK.

**Review by coignetp — APPROVED** [review]

**Pull request merged**